tdp/core/repository/git_repository.py

```diff
@@ -41,6 +46,13 @@ def init(path):
     def validate(self, msg):
         with self._lock:
             yield self
+            try:
+                if len(self._repo.index.diff("HEAD")) == 0:
+                    raise EmptyCommit("The commit has no diff")
+            except BadName as e:
+                logger.debug(
+                    f"error during diff: {e}. Probably because the repo is still empty."
+                )
             commit = self._repo.index.commit(msg)
             logger.info(f"commit: [{commit.hexsha}] {msg}")
```

Comment:
```suggestion
    def validate(self, msg):
        with self._lock:
            yield self
            try:
                diffs = self._repo.index.diff("HEAD")
            except BadName as e:
                logger.debug(
                    f"error during diff: {e}. Probably because the repo is still empty."
                )
                diffs = []
            if len(diffs) == 0:
                raise EmptyCommit("The commit has no diff")
            commit = self._repo.index.commit(msg)
            logger.info(f"commit: [{commit.hexsha}] {msg}")
```